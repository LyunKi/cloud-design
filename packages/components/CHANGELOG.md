# @cloud-design/components

## 0.2.0

### Minor Changes

- 增加 Text 实现

### Patch Changes

- Updated dependencies
  - @cloud-design/icons@0.2.0

## 0.1.0

### Minor Changes

- init cloud design components

### Patch Changes

- Updated dependencies
  - @cloud-design/icons@0.1.0
