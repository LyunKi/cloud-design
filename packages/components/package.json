{
  "name": "@cloud-design/components",
  "description": "React Native UI Components",
  "main": "./dist/index.js",
  "types": "./dist/index.d.ts",
  "files": [
    "dist"
  ],
  "scripts": {
    "build": "tsc -p ./tsconfig.build.json",
    "prepublish": "pnpm build",
    "test": "jest",
    "test:watch": "jest --watch ",
    "test:updateSnap": "jest --updateSnapshot"
  },
  "version": "0.2.0",
  "dependencies": {
    "lodash": "^4.17.21",
    "@cloud-design/icons": "0.2.0"
  },
  "peerDependencies": {
    "react": "^18.2.0",
    "react-native": "^0.69.4"
  },
  "devDependencies": {
    "react": "^18.2.0",
    "react-native": "^0.69.4",
    "@types/lodash": "^4.14.182",
    "@types/react": "^18.0.15",
    "@cloud-dragon/common-types": "^0.1.0"
  }
}
