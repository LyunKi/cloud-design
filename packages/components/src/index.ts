export * from './common'
export * from './View'
export * from './FlexLayout'
export * from './Text'
export * from './Button'
export * from './Icon'
export * from './Overlay'
export * from './Menu'
export * from './Divider'
export * from './ConfigProvider'
export * from './Avatar'
export * from './Input'
export * from './TopNavigation'
export * from './Layout'
