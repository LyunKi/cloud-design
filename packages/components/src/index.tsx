export * from '@cloud-design/icons'
export * from './ConfigProvider'
export * from './View'
export * from './FlexLayout'
