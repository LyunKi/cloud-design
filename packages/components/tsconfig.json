{
  "extends": "@cloud-dragon/tsconfig-react"
}
