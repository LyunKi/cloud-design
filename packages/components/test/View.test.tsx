import { render } from '@testing-library/react-native'
import { View } from '@cloud-design/components'
import React from 'react'

describe('<View />', () => {
  it('render correctly', () => {
    const snapshot = render(
      <View ts={{ color: '$color-primary-100' }} />
    ).toJSON()
    expect(snapshot).toMatchSnapshot()
  })
})
