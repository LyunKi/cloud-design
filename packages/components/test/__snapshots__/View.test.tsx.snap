// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<View /> render correctly 1`] = `
<View
  style={
    Object {
      "color": "#F2F6FF",
      "flexDirection": "row",
    }
  }
/>
`;
